### ISSUE TITLE
Variable product can be purchased without choosing variations with WooPay button on product page

### ISSUE THREAD

**Issue by LCmry** [Automattic/woocommerce-payments#7616]
### Describe the bug

When on a product page with the WooPay express button enabled, it is possible to purchase a variable product even when variations have not been selected. This would be a problem for merchants, because they would not know what variation was ordered.
### To Reproduce

1. Enable WooPay express checkout button on product page
2. Visit a page with a variable product and do not select any variations
3. Click the WooPay button and it will take you to WooPay checkout

### Actual behavior

The WooPay button continues to checkout without choosing variations for a variable product

### Expected behavior

A message should appear that lets the shopper know what they need to do - select variations - and prevents checkout. An example of this can be found on a Blocks product page. More info [here](https://github.com/Automattic/woocommerce-payments/issues/6634#issuecomment-1791431592).

### Additional context

Related to #6634 and #7158

**Comment by htdat** [comment]
This issue impacts "WooPay Express checkout", so assigning to Heisenberg (based on team responsibilities Pc2DNy-3z-p2) cc team lead @frosso. Assigning as part of Gamma Triage process PcreKM-yM-p2.

**Comment by bborman22** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/WooPay-621412506e2a6700140b7aac/issues/Automattic/woocommerce-payments/7616?planning-poker) with Zenhub @cesarcosta99 @lovo-h @rafaelzaleski @ricardo

**Comment by ricardo** [comment]
Rafael brought up a good point in this https://github.com/Automattic/woocommerce-payments/pull/7735#discussion_r1397312630 about the messaging used in the alert for other Payment Request Buttons when a variation is not selected:

> Please select some product options before adding this product to your cart.

Google Pay and Apple Pay use that message, which is default from WC Core, but it would be more concise if the message was something like this, which Rafael suggested:

> Configure your product options to proceed to payment.

@pierorocca Do you have any thoughts or suggestions on the right messaging? Should we also update the other PRBs messaging?

**Comment by pierorocca** [comment]
Is it possible to generate the message based on the attribute that hasn't been set?
e.g. Please select a size; Please select a size and color

cc. @elizaan36 and @nerrad 

-----
Longer term I'd love to see design and platform team move away from the system alerts approach. The button, inline messaging, and having default attributes on by default, would vastly improve the experience. As would moving away from drop-downs for the variation selector.

As an example, Allbird communicates so much through the dynamic button. https://d.pr/v/ZEYKJo

Alternatively Lululemon doesn't disable the button or use it as a way to communicate the CTA, they do use inline messaging to communicate instructions and visually where they should be applied. https://d.pr/v/J3wL7K

**Comment by pierorocca** [comment]
> Should we also update the other PRBs messaging?

Yes all must be identical if the button state is enabled at all times.

**Comment by pierorocca** [comment]
Also noting that of the sites I've looked at for inspiration, all force a default value for every attribute except for size. Size is the only attribute a shopper must select that isn't preselect for them.

**Comment by ricardo** [comment]
> Is it possible to generate the message based on the attribute that hasn't been set?
> e.g. Please select a size; Please select a size and color

The examples from these websites would indeed be more user-friendly, but I can see a couple of challenges and potential issues given WooCommerce is multi-language and highly customizable:
- The messaging might not combine well with pluralization of custom variable options. E.g: "Please select 'a' colors". While this might be easier to fix in English, some languages might require more specific use of grammar or pluralization which may not work well with a generic structure like this.
- If there's a custom option with a pretty long name that could be selected with "Yes/No", the user might get overwhelmed with information, or the messaging might not fit smaller screens. E.g: "Please select Custom Engraved Metallic Finish with Non-Slip Grip".
- In cases where multiple options must be selected in combination (like a bundle), it might also be challenging to construct a clear and concise message if there are numerous combinations of options.

One thing I forgot to mention in my initial question, is that this also applies for the compatibility with Mix and Match (#7364) and Product Bundles (#7158), where the product cannot be added to the cart before selecting all required options.

@pierorocca To unblock #7735, can we use some generic messaging like: "Please select all required options to proceed to payment." while we discuss the complexities of a more user-friendly approach?

**Comment by pierorocca** [comment]
> Should we also update the other PRBs messaging?
Yes all must be identical if the button state is enabled at all times.

Ok I understand the problem statement now. "to your cart" doesn't make sense for the PRBs. 

How about  "Please select your product options before proceeding."?

**Comment by elizaan36** [comment]
> Is it possible to generate the message based on the attribute that hasn't been set?
e.g. Please select a size; Please select a size and color

This makes perfect sense. Thanks for sharing those references - the Allbirds dynamic button is really nice but I think going in the Lululemon direction with inline messaging will be a good first step.

We could present a contextual notice above the selection that needs to be made.
<img width="1500" alt="Product - Vertical - More payment options" src="https://github.com/Automattic/woocommerce-payments/assets/5121465/223d38c6-22fe-4e1c-9d06-952aa9773232">

**Comment by ricardo** [comment]
This issue was automatically closed by #7735 which implements the same legacy alert used by other PRB, but do we have another issue in the WC Blocks or WC Core repository about the error notice?

**Comment by pierorocca** [comment]
> but do we have another issue in the WC Blocks or WC Core repository about the error notice?

I don't think there is a dedicated issue for this. I'll open it.
